### PR TITLE
Remove ignore of E731

### PR DESCRIPTION
**Pull request by horpto**

### CODE REVIEW

**Comment by menshikh-iv** [comment]
I close this PR, because PEP8 don't declare it properly (info in #1644)
Please switch to E501 if you have time (additional info in issue).

File: gensim/test/test_atmodel.py [hunk 1]
```diff
@@ -37,7 +37,10 @@
 # Test that models are compatiple across versions, as done in LdaModel.
 
 module_path = os.path.dirname(__file__)  # needed because sample data files are located in the same folder
-datapath = lambda fname: os.path.join(module_path, 'test_data', fname)
+
+
+def datapath(fname):
```
Comment (menshikh-iv) [inline gensim/test/test_atmodel.py]: please create `utils` file and move this function (replace to import everywhere).

File: gensim/test/test_fasttext_wrapper.py [hunk 2]
```diff
@@ -19,10 +19,13 @@
...
 module_path = os.path.dirname(__file__)  # needed because sample data files are located in the same folder
-datapath = lambda fname: os.path.join(module_path, 'test_data', fname)
 logger = logging.getLogger(__name__)
 
 
+def datapath(fname):
+    return os.path.join(module_path, 'test_data', fname)
+
+
 def testfile():
```
Comment (menshikh-iv) [inline gensim/test/test_fasttext_wrapper.py]: Same as `datapath`

File: gensim/test/test_ldamodel.py [hunk 3]
```diff
@@ -25,8 +25,6 @@
 from gensim.test import basetmtests
 
 module_path = os.path.dirname(__file__)  # needed because sample data files are located in the same folder
-datapath = lambda fname: os.path.join(module_path, 'test_data', fname)
-
 
 # set up vars used in testing ("Deerwester" from the web tutorial)
 texts = [
```
Comment (menshikh-iv) [inline gensim/test/test_ldamodel.py]: same as `datapath`

File: gensim/test/test_atmodel.py [hunk 4]
```diff
@@ -27,6 +24,7 @@
 from gensim.models import atmodel
 from gensim import matutils
 from gensim.test import basetmtests
+from gensim.test.utils import (datapath, get_tmpfile)
```
Comment (menshikh-iv) [inline gensim/test/test_atmodel.py]: useless `()` (here and anywhere)

File: gensim/test/test_atmodel.py [hunk 5]
```diff
@@ -507,7 +498,7 @@ def testPersistence(self):
         self.assertTrue(np.allclose(model.state.gamma, model2.state.gamma))
 
     def testPersistenceIgnore(self):
-        fname = testfile('testPersistenceIgnore')
+        fname = get_tmpfile('gensim_models_atmodel_testPersistenceIgnore.tst')
```
Comment (menshikh-iv) [inline gensim/test/test_atmodel.py]: it's a good idea to use same notation, what's your choice?

File: gensim/test/test_atmodel.py [hunk 6]
```diff
@@ -36,9 +34,6 @@
 # increases the bound.
 # Test that models are compatiple across versions, as done in LdaModel.
 
-module_path = os.path.dirname(__file__)  # needed because sample data files are located in the same folder
-datapath = lambda fname: os.path.join(module_path, 'test_data', fname)
-
 # set up vars used in testing ("Deerwester" from the web tutorial)
 texts = [
```
Comment (menshikh-iv) [inline gensim/test/test_atmodel.py]: This `text` used in many places, maybe move to utils too?

File: gensim/test/utils.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+"""
+Common utils for tests
+"""
+import tempfile
+import os
+
+module_path = os.path.dirname(__file__)  # needed because sample data files are located in the same folder
+
+
+def datapath(fname):
```
Comment (menshikh-iv) [inline gensim/test/utils.py]: Please add docstring in [numpy-style](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt) for both functions

File: gensim/utils.py [hunk 8]
```diff
@@ -336,13 +337,8 @@ def _load_specials(self, fname, mmap, compress, subname):
...
     def _adapt_by_suffix(fname):
         """Give appropriate compress setting and filename formula"""
-        if fname.endswith('.gz') or fname.endswith('.bz2'):
-            compress = True
-            subname = lambda *args: '.'.join(list(args) + ['npz'])
-        else:
-            compress = False
-            subname = lambda *args: '.'.join(list(args) + ['npy'])
-        return compress, subname
+        [compress, suffix] = (True, 'npz') if fname.endswith('.gz') or fname.endswith('.bz2') else (False, 'npy')
```
Comment (menshikh-iv) [inline gensim/utils.py]: useless `[]`

File: gensim/test/test_atmodel.py (hunk 6 above)
Comment (horpto) [reply gensim/test/test_atmodel.py]: hmm, 
1) it's data, not utils/functions
2) maybe in another PR ?

Comment (menshikh-iv) [reply gensim/test/test_atmodel.py]: 1. Yeah, but it's "common resourse", I think that's fine.
2. No need to create distinct PR for this change.

Comment (horpto) [reply gensim/test/test_atmodel.py]: 1. Ok, I'll fix it.
2. small PR is better to review and faster to merge.

**Comment by menshikh-iv** [comment]
close/open for init AppVeyor

**Pull request merged**